src/cla/tcp/net.rs

```diff
@@ -75,26 +94,53 @@ impl TcpClPacket {
                 MessageType::XferSegment => {
                     let flags = XferSegmentFlags::from_bits_truncate(reader.read_u8().await?);
                     let tid: u64 = reader.read_u64().await?;
+                    let mut extensions = Vec::new();
                     if flags.contains(XferSegmentFlags::START) {
-                        let ext_len: u32 = reader.read_u32().await?;
+                        let mut ext_len: u32 = reader.read_u32().await?;
+                        // parse bundle ids that are request
                         if ext_len != 0 {
-                            warn!("transfer extension are unsupported, ignoring them!");
-                        }
-                        for _ in 0..ext_len {
-                            reader.read_u8().await?;
+                            debug!("parsing transfer extensions");
+                            while ext_len >= 3 {
+                                let flag = reader.read_u8().await?;
+                                let item_type = reader.read_u16().await?;
+                                let item_length = reader.read_u16().await?;
+                                ext_len = ext_len - 3 - item_length as u32;
+                                let mut data = Vec::with_capacity(item_length as usize);
+                                data.resize_with(item_length as usize, Default::default);
+                                reader.read_exact(&mut data).await?;
+                                if let Some(item_type) =
+                                    TransferExtensionItemType::from_u16(item_type)
+                                {
+                                    let transfer_extension = TransferExtensionItem {
+                                        flags: TransferExtensionItemFlags::from_bits_truncate(flag),
+                                        item_type,
+                                        data: data.into(),
+                                    };
+                                    extensions.push(transfer_extension);
+                                }
+                            }
+                            if ext_len != 0 {
+                                warn!("malformed transfer extensions, ignoring rest");
+                                for _ in 0..ext_len {
+                                    reader.read_u8().await?;
+                                }
+                            }
                         }
                     }
                     let len = reader.read_u64().await?;
                     debug!("Reading xfer segment with len {}", len);
                     let mut data = Vec::with_capacity(len as usize);
-                    data.resize_with(len as usize, Default::default);
-                    reader.read_exact(&mut data).await?;
+                    if len > 0 {
+                        data.resize_with(len as usize, Default::default);
```

Comment:
from https://doc.rust-lang.org/std/vec/struct.Vec.html:
> It can also initialize each element of a Vec<T> with a given value. This may be more efficient than performing allocation and initialization in separate steps, especially when initializing a vector of zeros:
```rust
let vec = vec![0; 5];
assert_eq!(vec, [0, 0, 0, 0, 0]);

// The following is equivalent, but potentially slower:
let mut vec = Vec::with_capacity(5);
vec.resize(5, 0);
assert_eq!(vec, [0, 0, 0, 0, 0]);
```